### PR TITLE
ID-1091 Make Sam publish client with SemVer versioning

### PR DESCRIPTION
**Pull request by tlangs**
Ticket: https://broadworkbench.atlassian.net/browse/ID-1091

Make Sam's Client publish with the sam SemVer as Sam itself!

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've filled out the [Security Risk Assessment](https://sdarq.dsp-appsec.broadinstitute.org/jira-ticket-risk-assesment) (requires Broad Internal network access) and attached the result to the JIRA ticket

### CODE REVIEW

**Comment by tlangs** [comment]
Check out the publish steps here: https://github.com/broadinstitute/sam/actions/runs/8114255218/job/22179650378?pr=1370

File: codegen_java/project/Version.scala [hunk 1]
```diff
@@ -11,10 +11,12 @@ object Version {
...
     val version = baseVersion + "-" + lastCommit
 
+    val semVer = sys.props.get("project.semVer").getOrElse(version)
+
     // The project isSnapshot string passed in via command line settings, if desired.
     val isSnapshot = sys.props.getOrElse("project.isSnapshot", "true").toBoolean
 
     // For now, obfuscate SNAPSHOTs from sbt's developers: https://github.com/sbt/sbt/issues/2687#issuecomment-236586241
-    if (isSnapshot) s"$version-SNAP" else version
+    if (isSnapshot) s"$semVer-SNAP" else semVer
```
Comment (okotsopoulos) [inline codegen_java/project/Version.scala:20]: Will you help clarify for me -- when are SNAPSHOTs published?  Should downstream consumers be ignoring them generally (thinking of Dependabot rules).

**Review by okotsopoulos — APPROVED** [review]
Thanks very much for this!

File: codegen_java/project/Version.scala (hunk 1 above)
Comment (tlangs) [reply codegen_java/project/Version.scala:20]: `-SNAP` versions are published for unmerged code. When code is merged to develop, it'll just be the SemVer version without the `-SNAP` suffix.

Comment (okotsopoulos) [reply codegen_java/project/Version.scala:20]: Cool, that's what I thought but I appreciate the confirmation.

**Review by dvoet — APPROVED** [review]

**Comment by tlangs** [comment]
No code changes made. Merging.

**Pull request merged**